implementations/elixir/ockam/ockam_services/test/services/app_test.exs

```diff
@@ -6,13 +6,16 @@ defmodule Test.Services.AppTest do
   test "app runs configured services" do
     Application.put_env(:ockam_services, :service_providers, [Test.Services.AppTest.Provider])
     Application.put_env(:ockam_services, :services, [:my_service])
-    Application.put_env(:ockam_services, :services_config_source, nil)
 
     {:ok, _apps} = Application.ensure_all_started(:ockam_services)
 
-    sup = Ockam.Services
+    on_exit(fn ->
+      Application.delete_env(:ockam_services, :service_providers)
+      Application.delete_env(:ockam_services, :services)
+      Application.stop(:ockam_services)
```

Comment:
I think this may behave oddly if this test module appears in the middle of the sequence for the test suite (or especially if the module was ever marked async). Is there a less extreme cleanup step we can take instead, or can we at least start this OTP app right back up as the next line? Maybe then we wouldn't need the `ensure_all_started`.

Reply:
In the rest of this app tests the app is not supposed to be running and this test is testing the start sequence of the app. I think that stopping it as a cleanup step in that context makes sense.